api/src/identity-access-management/infrastructure/repositories/authentication-method.repository.js

```diff
@@ -269,6 +270,33 @@ const batchUpdatePasswordThatShouldBeChanged = function ({
   );
 };
 
+/**
+ * @param {array} userIds
+ * @param {number} chunkSize
+ * @returns {Promise<{linesUpdated: number}>}
+ */
+const batchAnonymizeByUserIds = async function ({ userIds, chunkSize = 1000 }) {
+  const userIdBatches = _.chunk(userIds, chunkSize);
+  let anonymizedUserIds = 0;
+
+  await bluebird.mapSeries(userIdBatches, async (userIdBatch) => {
+    const anonymizedUserIdBatch = await knex(AUTHENTICATION_METHODS_TABLE)
+      .whereIn('userId', userIdBatch)
+      .andWhere('identityProvider', 'GAR')
+      .update(
+        {
+          authenticationComplement: { firstName: 'anonymized', lastName: 'anonymized' },
+          updatedAt: knex.fn.now(),
+          externalIdentifier: knex.raw('CONCAT(\'anonymized-\', "authentication-methods".id)'),
+        },
+        [COLUMNS],
+      );
+    anonymizedUserIds += anonymizedUserIdBatch.length;
+  });
+
+  return { linesUpdated: anonymizedUserIds };
```

Comment:
Je trouve un peu perturbant les noms de propriété `linesUpdated` et `anonymizedUserIds`. 
`linesUpdated` semble vouloir dire qu'on va y trouver toutes les lignes modifiées.
`anonymizedUserIds` semble vouloir dire qu'on va avoir la liste de tous les ID modifiés.
Aussi ci-dessous je propose un nommage que j'espère plus explicite : 

```suggestion
  let anonymizedUserCount = 0;

  await bluebird.mapSeries(userIdBatches, async (userIdBatch) => {
    const anonymizedUserIdBatch = await knex(AUTHENTICATION_METHODS_TABLE)
      .whereIn('userId', userIdBatch)
      .andWhere('identityProvider', 'GAR')
      .update(
        {
          authenticationComplement: { firstName: 'anonymized', lastName: 'anonymized' },
          updatedAt: knex.fn.now(),
          externalIdentifier: knex.raw('CONCAT(\'anonymized-\', "authentication-methods".id)'),
        },
        [COLUMNS],
      );
    anonymizedUserCount += anonymizedUserIdBatch.length;
  });

  return { anonymizedUserCount };
```